lib/slashdeploy/service.rb

```diff
@@ -34,18 +34,20 @@ def update_repository_config(repository)
     #
     # Returns an AutoDeployment.
     def create_auto_deployment(environment, sha, user)
-      auto_deployment = environment.auto_deployments.create! user: user, sha: sha
-      state = auto_deployment.state
-      case state
-      when AutoDeployment::STATE_READY
-        auto_deploy auto_deployment
-      when AutoDeployment::STATE_PENDING
-        direct_message \
-          auto_deployment.slack_account, \
-          AutoDeploymentCreatedMessage, \
-          auto_deployment: auto_deployment
-      else
-        fail "Unhandled #{state} state"
+      auto_deployment = environment.auto_deployments.create user: user, sha: sha
+      if auto_deployment.valid?
```

Comment:
Change to:

```ruby
return auto_deployment unless valid?
```

And remove the nested if block.

Reply:
cool. I have a habit of trying to limit the amount of returns a function performs, especially if I intend to return a specific type.